tree/path.go

```diff
@@ -51,6 +51,16 @@ func (p Path) Parts() []string {
 	return strings.Split(string(p), pathSeparator)
 }
 
+func (p Path) ContainsPart(part string) bool {
```

Comment:
This could work for this use case but is a bit fragile.
IMHO we could make this more deterministic if `Matches` adds support for "anything and child" `**` syntax, so that you can use :
```
KeysToInterpolate := []{
  "services.*.extends.**"
  "includes.**"
}
```

Reply:
pushed changes, wrote tests, if you know any other cases that can be useful, let me know